src/uas/UASManagerInterface.h

```diff
@@ -98,6 +98,8 @@ public slots:
     void UASCreated(UASInterface* UAS);
     /** A system was deleted */
     void UASDeleted(UASInterface* UAS);
+    /** A system was deleted */
+    void UASDeleted(int systemId);
```

Comment:
I've been trying to get rid of all the overridden signals with same name different params. The reason is that it the forces you to use the SIGNAL/SLOT connect mechanism instead of the direct method reference . The SIGNAL/SLOT connect mechanism doesn't provide compile time checking for incorrect usage. There were a pile of problems caused by not having this compile time check with broken signals. Can't the HIL has map be changed to index to pointer instead of id?